### DISCUSSION TITLE
Add Color Picker option to pick color through transparent pixel (as if no pixel at all)

### DISCUSSION THREAD

**Discussion by hsandt** [Orama-Interactive/Pixelorama#998] [Ideas]
Currently, if you paint with transparent pixels (using a color whose alpha = 0), it will store transparent pixels on the layer, which are different from having no pixels at all (e.g. on a new layer or after erasure).

When using Color Picker in Pick mode: Top Color and a top layer contains transparent pixels over other layers containing some color C, when clicking on a transparent pixel, the transparent color will be selected. While there may be some edge cases where it's useful (e.g. to find out which hue was supposed to be there if it wasn't completely transparent, or force-marking an area as "not painted"), in most cases I'd expect the top color to be the first **non-fully-transparent** color encountered when checking layers from top to bottom.

In addition, it is common to use transparent pixels as a way of erasure (for instance to fill an area with the Bucket) and there is no visible distinction, so I would expect both to behave the same with the Color Picker.

So, to focus on the first non-fully-transparent pixel while preserving the old behavior for edge cases, I would add an option "ignore fully transparent pixels" (only visible when Pick mode is Top color) that is checked by default.

EDIT: Alternatively, there could be an option to always erase pixels and never store "fully transparent pixels".

EDIT 2: I opened a bug too because behavior could be considered like a bug by users, although technically it does make sense: https://github.com/Orama-Interactive/Pixelorama/issues/999

**Comment by OverloadedOrama** [comment]
I think it's probably best to consider this a bug, so I'll close this discussion as a duplicate of #999. I'm not sure if there's any strong reason to keep the current behavior as an option, but of course feel free to let me know if this is something potentially useful for you.
